src/brogue/Config.c

```diff
@@ -0,0 +1,255 @@
+    long fileSize = ftell(jsonFile);
+    fseek(jsonFile, 0, SEEK_SET);
+
+    char* buffer = (char*)malloc(fileSize + 1);
+
+    if (!buffer) {
+        fclose(jsonFile);
+        return NULL;
+    }
+
+    fread(buffer, 1, fileSize, jsonFile);
+    fclose(jsonFile);
+
+    buffer[fileSize] = '\0';
+
+    return buffer;
+}
+
+static configField* getFieldEntries(configParams* config) {
+    int numFields = 7;
+
+    configField fieldDescriptors[] = {
+        {"gameVariant", &(config->gameVariant), ENUM_STRING, {.enumMappings = variantMappings}},
+        {"graphicsMode", &(config->graphicsMode), ENUM_STRING, {.enumMappings = graphicsModeMappings}},
+        {"playbackDelayPerTurn", &(config->playbackDelayPerTurn), INT_TYPE, {.intRange = playbackDelayRange}},
+        {"displayStealthRangeMode", &(config->displayStealthRangeMode), BOOLEAN_TYPE},
+        {"trueColorMode", &(config->trueColorMode), BOOLEAN_TYPE},
+        {"wizard", &(config->wizard), BOOLEAN_TYPE},
+        {"easyMode", &(config->easyMode), BOOLEAN_TYPE},
+        {NULL}};
+
+    configField* entries = calloc(numFields + 1, sizeof(configField));
+
+    for (int i = 0; i < numFields; i++) {
+        entries[i] = fieldDescriptors[i];
+    }
+
+    return entries;
+}
+
+static short mapStringToEnum(const char* inputString, const char** mappings) {
+    for (short i = 0; mappings[i]; i++) {
+        if (strcmp(inputString, mappings[i]) == 0) {
+            return i;
+        }
+    }
+    return -1;
+}
+
+static void parseConfigValues(const char* jsonString, configParams* config) {
+    if (!jsonString || !config) {
+        return; // Invalid input
+    }
+
+    cJSON* root = cJSON_Parse(jsonString);
+
+    if (!root) {
+        return; // JSON parsing error
+    }
+
+    configField* entries = getFieldEntries(config);
+
+    for (int i = 0; entries[i].name; i++) {
+        cJSON* jsonField = cJSON_GetObjectItem(root, entries[i].name);
+
+        if (jsonField) {
+            switch (entries[i].type) {
+            case INT_TYPE:
+                if (cJSON_IsNumber(jsonField)) {
+                    short value = jsonField->valueint;
+                    intRange valueRange = entries[i].validator.intRange;
+
+                    if (value >= valueRange.min && value <= valueRange.max) {
+                        *((short*)entries[i].paramPointer) = value;
+                    }
+                }
+                break;
+
+            case BOOLEAN_TYPE:
+                if (cJSON_IsBool(jsonField)) {
+                    *((boolean*)entries[i].paramPointer) = jsonField->valueint;
+                }
+                break;
+
+            case ENUM_STRING:
+                if (cJSON_IsString(jsonField)) {
+                    const char* modeString = jsonField->valuestring;
+                    short mode = mapStringToEnum(modeString, entries[i].validator.enumMappings);
+
+                    if (mode != -1) {
+                        *((short*)entries[i].paramPointer) = mode;
```

Comment:
I'm not sure if all of the enum types are definitely short-sized. It would be nice to have some better type-safety here, maybe a constructor function for enum struct field definitions that explicitly only accepts `short*`?